### ISSUE TITLE
pfSense wazuh agent doesn't collect local log file

### ISSUE THREAD

**Issue by Evacch** [wazuh/wazuh#14539]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.7 | localfile configuration | Agent | FreeBSD pkg | FreeBSD |

I'm trying to integrate pfSense Suricata with Wazuh by referring the tutorial in https://benheater.com/integrating-pfsense-with-wazuh/.
Everything was working fine, wazuh-agent service is running and it was also configured to read suricata log in Wazuh, 
![image](https://user-images.githubusercontent.com/76742978/184050971-f9721315-2244-4b0f-9f9f-390230237cd3.png)

pfSense was successfully connected to Wazuh and there is suricata log in pfSense but suricata log doesn't show in the Wazuh dashboard.

In wazuh agent(pfSense) ossec.log, I observed the following error messages.
![image](https://user-images.githubusercontent.com/76742978/184050454-7697a0ae-76bf-456d-8190-99da553da1cb.png)

I am not sure if this is the issue that causes suricata log unable to sync with Wazuh.

**Comment by jnasselle** [comment]
Hi @Evacch,

There are a few things to ask before continuing to analyze this

- Tutorial use v4.3 version, but apparently you are using v4.2.7. It's this correct? did you apply the ownership changes mentioned in the tutorial(BTW, this is not recommended)
- Could you please validate that on `/var/log/suricata/*/eve.json` log file exist? AFAIK logs are stored directly on suricata's `default-log-dir` configuration path. Proper localfile configuration should use `json` as `log_format`

Also, there's a nice article on the Wazuh documentation about this https://documentation.wazuh.com/current/learning-wazuh/suricata.html

**Comment by 0xBEN** [comment]
@jnasselle [when I first wrote about ingesting pfSense logs](https://benheater.com/integrating-pfsense-with-wazuh/) (including Suricata) with Wazuh, I was testing on version `4.2.x`. pfSense logs Suricata a bit differently than a default install on a Linux server.

pfSense logs Suricata in a /var/log/suricata/<interface-name.identifier>/eve.json` hierarchy. Therefore, to ingest multiple logs, the `/var/log/suricata/*/eve.json` pattern is used. In version `4.2.x` of the Wazuh agent, the file globbing worked just fine and all of the logs were ingested.

However, after upgrading the agent to `4.3.x`, the file globbing stopped working. Upon checking the agent logs, there were references to permissions issues.
```
2022/08/13 00:01:04 wazuh-agentd: WARNING: Unknown message received. No action defined.
2022/08/13 00:01:14 wazuh-agentd: ERROR: (1103): Could not open file 'etc/shared/merged.mg' due to [(13)-(Permission denied)].
```
It's not that the file globbing wasn't working, it's that the agent couldn't read the shared configuration pushed by the Wazuh manager and therefore, was not ingesting the log files. 

I understand that the recommended fix: `chown -R wazuh:wazuh /var/ossec` might not be optimal, but it's the only thing I could think to do at the time, since it wasn't clear at what level I'd need to modify ownership to get the shared configuration files to be pulled in by the agent. I just didn't know to what extent the permissions issue was affecting the agent's capabilities. If you have a better fix, please do suggest it, so I can update my documentation.

Long story short, somewhere between `4.2.x` and `4.3.x`, file permissions changes became broken and the agent could read the shared configuration, thus `<localfile>` directives were not adopted by the agent.

**Comment by 0xBEN** [comment]
This is on Wazuh Agent `4.3.7`, after a fresh reinstall. The Wazuh Manager is at `4.3.7-1`. I uninstalled the Wazuh agent on pfSense, ran `rm -rf /var/ossec`, killed any lingering processes.

Then, reinstalled with `pkg install wazuh-agent` . The agent installed successfully. After that, I made `0` modifications to the file system. I added the `pfSense` agent to the group to adopt the shared configuration. The shared configuration failed to apply. After looking at the logs, this is what I see.

![image](https://user-images.githubusercontent.com/26947869/188017356-866ae0ee-b053-4d91-ae7d-89397dc011a8.png)

The shared configuration files are completely absent on the host.

![image](https://user-images.githubusercontent.com/26947869/188017511-a4cf108f-49bb-4e35-9ddf-6f8a572b1828.png)

**Comment by 0xBEN** [comment]
I also wanted to share the Suricata logging hierarchy as well.

![image](https://user-images.githubusercontent.com/26947869/188024325-7748319c-5ad6-4e3f-904b-30a29fbb5481.png)

Let me know if I can provide any additional details.

**Comment by 0xBEN** [comment]
Also, here's another detail. This is the `/var/ossect/etc` directory on a Linux server with agent version `4.3.3`
![image](https://user-images.githubusercontent.com/26947869/188025237-a5dd16a8-7217-4376-9339-0c85e80fa4b0.png)

This is the `/var/ossec/etc/` directory on pfSense (FreeBSD). 

![image](https://user-images.githubusercontent.com/26947869/188025337-9527ca45-4746-4572-a0cc-264a6bae113a.png)

You'll note that the `shared` directory on the Linux server is set to `770` and the `shared` directory on the pfSense host is set to `750`.

**Comment by alonsobsd** [comment]
Hi, I'm a maintainer of wazuh port/package on FreeBSD. I think this issue was resolved. It was my fault with some directory permissions when it is installed. I have updated it to 4.3.10 right now.

**Comment by 0xBEN** [comment]
Thanks for the update @alonsobsd and for your work maintaining the package. I'll spin up a test instance of pfSense and give it a test as soon as I can.

**Comment by balvinp** [comment]
Hi, I wanted to try installing the wazuh-agent package on pfsense 2.6 but it seems that its failing to build and when I try pkg search wazuh-agent I dont get any available packages to install. 

I checked [ports fallout](https://portsfallout.com/fallout?port=wazuh-agent%24&maintainer=&env=amd64&category=&flavor=) and the log for the last build attempt shows [missing nghttp2 library](https://pkg-status.freebsd.org/beefy6/data/123amd64-default/387f99b7180f/logs/wazuh-agent-4.3.9.log).

Wanted to compile it myself but didn't find freebsd folder in wazuh-packages repo.

**Comment by 0xBEN** [comment]
@alonsobsd I am happy to report that I was able to test the installation of `wazuh-agent-4.3.10` on a pfSense host in my lab and that there are no issues with the ingestion of log files due to permissions issues.

@balvinp not sure it it's coincidence or if the package maintainers fixed the build issues after seeing your comment, but the package is available for installation from the BSD repos on the latest version of pfSense.

**Comment by alonsobsd** [comment]
@balvinp if you want to compile it you must update your ports tree or change pkg mirror to latest if you want use packages like @0xBEN said you

@0xBEN good to read it. Thanks for your test

**Comment by alonsobsd** [comment]
Could you close this issue? It was fixed from fbsd side and I think it is not a issue with wazuh

**Comment by 0xBEN** [comment]
@Evacch please consider closing this issue if you feel it has been fixed.
